Source/src/Gameplay.h

```diff
@@ -413,4 +414,51 @@ HACHIKO_API GameObject* Raycast(const float3& origin,
 namespace Hachiko::Debug
 {
 HACHIKO_API const Hachiko::ComponentCamera* GetEditorCamera();
-} // namespace Hachiko::Debug
\ No newline at end of file
+} // namespace Hachiko::Debug
+
+
+#define TYPE_IF_DERIVED_CLASS(BASE, DERIVED, TYPE)              \
+    std::enable_if_t<std::is_base_of<BASE, DERIVED>::value, TYPE>
+
+#define HACHIKO_API_COMPONENT_VOID                                \
+    template<class COMPONENT_TYPE>                                \
+    HACHIKO_API                                                   \
+    TYPE_IF_DERIVED_CLASS(Hachiko::Component, COMPONENT_TYPE, void)
+
+namespace Hachiko::Editor
+{
+HACHIKO_API void ShowGameObjectDragDropArea(const char* field_name, 
+    const char* field_type, GameObject** game_object, bool& changed);
```

Comment:
why are we getting the bool by reference? it makes no difference to get it by value, since the value and the reference pretty much are the same size, but will cause the caller to have to declare a bool variable to pass it on

Reply:
The reason we have `changed` as pass by ref is that it actually overrides that changed and it can be used by the caller for any reason that requires to see if the `*game_object` was changed. I made this function that way to explicitly show that function gives a `bool` that depicts a change, instead of returning a `bool` which didn't made a sense with a function named like that. If you think we should change this function's return type to `bool` and delete `changed` parameter, I can do that.

Reply:
With a name like that you'd also not expect that a parameter changes, I guess we can leave it as is

Reply:
Yeah you are right, let's leave it like that for now until we find a better name.